pkg/lib/operation/project/local/template/use/operation.go

```diff
@@ -78,12 +78,27 @@ func Run(projectState *project.State, tmpl *template.Template, o Options, d depe
 		return err
 	}
 
-	// Rename and save
+	// Prepare operations
 	objects := make(newObjects, 0)
 	errors := utils.NewMultiError()
 	renameOp := projectState.LocalManager().NewPathsGenerator(true)
 	saveOp := projectState.LocalManager().NewUnitOfWork(projectState.Ctx())
+
+	// Store template information in branch metadata
+	branchState := projectState.GetOrNil(o.TargetBranch).(*model.BranchState)
+	version := tmpl.Version()
+	if err := branchState.Local.Metadata.AddTemplateUsage(instanceId, tmpl.TemplateId(), version.String()); err != nil {
+		errors.Append(err)
+	}
+	saveOp.SaveObject(branchState, branchState.LocalState(), model.NewChangedFields())
```

Comment:
@JakubMatejka , sorry, ze som hovoril o `persist`, ono to je potrebne pridat do `save` operacie.
Tak som to rovno presunul, zvysok idem este pozriet.